internal/provider/schema_gitlab_project_cluster_agent.go

```diff
@@ -0,0 +1,50 @@
+package provider
```

Comment:
Hm... new approach for the schema separating this off into a new file. I assume this is done to prevent a lot of code duplication between the data source and the resource? It feels like we may be hitting a critical mass where we want another package for the schema files (and maybe helpers?), just so we don't have [schema, resource, resource tests, data sources, data source tests, helpers] all in one package.

Reply:
Yes, it's exactly about reducing code duplication. I'm not sure about the packages in this case. What benefits would it bring besides "the critical mass" (which IMHO is not a very good reason alone - convince me otherwise, please 💃 ) ?

Reply:
It would have to be for productivity reasons, right? For me personally I am ok with at least having the resources, data sources, schemas, and resource tests in the same package because the filenames make it easy to navigate. I could see wanting to move our various helpers and "main" provider code into new package/s though, since those can be hard to locate.

Reply:
@timofurrer  - Yeah, as adam says it's more around making it easy to find what you're looking for. Part of me kinda wishes we did <resource_name>-(resource|data source|schema) because then they'd all be grouped together when you were developing one of them, but I suspect that would just cause other things to be difficult to find.

I'm just a bit worried that with the number of files in the package it may get unweildy for people to find what they need to work on. The counter argument to this is of course... that's what an IDE is for. Ctrl+Click and find file make locating files easier. So I'm not passionate about the commentary, it was just something that went through my mind when I was reviewing and figured it was worth leaving a comment to see what you two thought. 

I do like the schema pattern though, so no issues there!

Reply:
I've created https://github.com/gitlabhq/terraform-provider-gitlab/issues/1083 as a follow up of this conversion. Let's continue the discussion there :)